image/preseed/preseed_linux.go

```diff
@@ -242,7 +242,23 @@ func prepareCore20Mountpoints(opts *preseedCoreOptions) (cleanupMounts func(), e
 		}
 	}
 
+	underWritable := func(path string) string {
+		return filepath.Join(opts.WritableDir, path)
+	}
+
+	currentLink := underWritable("system-data/snap/snapd/current")
```

Comment:
this could use a little comment for posterity, eg.
```
// because of the way snapd snap is built, we need the 'current' symlink
// to exist when the snapd binary is invoked, so that any runtime libraries
// will be correctly resolved, so bind-mount a snapd snap at a side location
// set up 'current' as symlink to it, which can then be easily replaced by 
// snapd as it preseeds the image